src/core/config/config-manager.ts

```diff
@@ -64,3 +68,328 @@ export class ConfigManager {
+        return true;
+    }
+
+    /**
+     * Delete a configuration
+     */
+    async deleteConfiguration(id: string): Promise<boolean> {
+        await this.initialize();
+
+        const manifest = await this.loadManifest();
+        const configMeta = manifest.configurations[id];
+
+        if (!configMeta) {
+            return false;
+        }
+
+        const configPath = path.join(this.configsDir, `${id}.yml`);
+
+        try {
+            // Remove the YAML file
+            await fs.unlink(configPath);
+
+            // Update manifest
+            delete manifest.configurations[id];
+            await this.saveManifest(manifest);
+
+            logger.info(`Configuration '${configMeta.name}' deleted`);
+            return true;
+        } catch (error) {
+            logger.error(`Failed to delete configuration ${id}: ${error}`);
+            return false;
+        }
+    }
+
+    /**
+     * Search configurations by name, description, or tags
+     */
+    async searchConfigurations(query: string): Promise<Omit<SavedConfiguration, 'config'>[]> {
+        const configurations = await this.listConfigurations();
+        const lowercaseQuery = query.toLowerCase();
+
+        return configurations.filter(
+            (config) =>
+                config.name.toLowerCase().includes(lowercaseQuery) ||
+                config.description.toLowerCase().includes(lowercaseQuery) ||
+                config.tags.some((tag) => tag.toLowerCase().includes(lowercaseQuery))
+        );
+    }
+
+    /**
+     * Get configurations by tags
+     */
+    async getConfigurationsByTags(tags: string[]): Promise<Omit<SavedConfiguration, 'config'>[]> {
+        const configurations = await this.listConfigurations();
+        return configurations.filter((config) => tags.some((tag) => config.tags.includes(tag)));
+    }
+
+    /**
+     * Export a configuration to a file
+     */
+    async exportConfiguration(id: string, targetPath: string): Promise<boolean> {
+        const config = await this.loadConfiguration(id);
+        if (!config) {
+            return false;
+        }
+
+        try {
+            const yamlContent = YAML.stringify(config.config, { lineWidth: -1 });
+            await fs.writeFile(targetPath, yamlContent);
+            logger.info(`Configuration '${config.name}' exported to ${targetPath}`);
+            return true;
+        } catch (error) {
+            logger.error(`Failed to export configuration: ${error}`);
+            return false;
+        }
+    }
+
+    /**
+     * Import a configuration from a file
+     */
+    async importConfiguration(
+        filePath: string,
+        name: string,
+        description: string,
+        tags: string[] = []
+    ): Promise<string | null> {
+        try {
+            const yamlContent = await fs.readFile(filePath, 'utf-8');
+            const config = YAML.parse(yamlContent) as AgentConfig;
+
+            return await this.saveConfiguration(name, description, config, tags);
```

Comment:
i don't understand this function

it loads from a file and then calls saveConfiguration which writes to a new file

is this the intention?

we should make the docstrings better for each of the functions, they don't help explain what is going on